crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -150,6 +150,22 @@ def compute_resources_available?
       true
     end
 
+    # Check for presence of HA setup, which is a requirement for non-disruptive upgrade
+    def ha_deployed?
```

Comment:
do we want to name that more specific, like e.g. control_plane_on_ha (or something like that)? I guess eventually we want to check that the relevant roles are deployed properly on a ha cluster like we expect them to be, and not only whether a proposal exists. 

Do we also want to return not only a boolean result but also a error message, like in a status code + error message combination? this could be displayed to the user then.